smart-send-logistics/includes/frontend/class-ss-shipping-endpoint.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+if (!defined('ABSPATH')) {
+    exit; // Exit if accessed directly
+}
+include '../../pickup-point-block/smart-send-utility-session.php';
+/**
+ * Class SS_Shipping_Api_Endpoint
+ *
+ * Handles the REST API endpoints for Smart Send Logistics.
+ *
+ * This class is responsible for registering the REST API endpoints and handling
+ * requests to fetch the closest pickup points based on provided address parameters.
+ *
+ * @package    SmartSendLogistics
+ * @subpackage SS_Shipping_Endpoint
+ * @category   REST API
+ * @author     Smart Send
+ */
+class SS_Shipping_Api_Endpoint
+{
+
+    /**
+     * @var FrontendInterface $frontend An instance of the frontend class handling agent fetching and session management.
+     */
+    protected $frontend;
+
+    /**
+     * SS_Shipping_Endpoint constructor.
+     *
+     * @param SS_Shipping_Frontend $frontends An instance of a class implementing the frontend logic.
+     */
+    public function __construct(SS_Shipping_Frontend $frontends)
+    {
+        $this->frontend = $frontends;
+        add_action('rest_api_init', array($this, 'register_endpoints'));
+    }
+
+    /**
+     * Registers the custom REST API endpoint for fetching closest pickup points.
+     *
+     * @return void
+     */
+    public function register_endpoints()
+    {
+        // Register the API endpoint
+        register_rest_route('smart-send-logistics/v1', '/get-closest-pickup-points', array(
+            'methods' => 'POST',
+            'callback' => array($this, 'get_pickup_points_callback'),
+            'permission_callback' => "__return_true",
+        ));
+        // Register the API endpoint for shipping carrier
+        register_rest_route('smart-send-logistics/v1', '/get-shipping-carrier', array(
+            'methods' => 'POST',
+            'callback' => array($this, 'get_shipping_carrier'),
+            'permission_callback' => "__return_true",
+        ));
```

Comment:
These are not the Rest API endpoint dicussed on September 11th. Why are these used?